homeassistant/components/insteon/__init__.py

```diff
@@ -464,6 +72,14 @@ def _fire_button_on_off_event(address, group, val):
 
     insteon_modem = conn.protocol
 
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN]["modem"] = insteon_modem
+    hass.data[DOMAIN][INSTEON_ENTITIES] = {}
+
+    register_new_device_callback(hass, config, insteon_modem)
+    hass.async_add_job(register_services, hass, config, insteon_modem)
```

Comment:
Why do we need to add a job to do this? I think we can just call this, if we change `register_services` to an async safe callback.